syntax = "proto3";

service GithubSearchService {
    rpc Search(SearchRequest) returns (SearchResponse);
}

message SearchRequest {
    string search_term = 1;
    map<string, string> qualifiers = 2;
}

message SearchResponse {
    repeated Result results = 1;
}

message Result {
    string file_url = 1;
    string repo = 2;
}
